libmscore/part.cpp

```diff
@@ -539,6 +539,60 @@ int Part::harmonyCount() const
       return count;
       }
 
+//---------------------------------------------------------
+//   updateHarmonyChannels
+///   update the harmony channel by creating a new channel
+///   when appropriate or using the existing one
+///
+///   checkRemoval can be set to true to check to see if we
+///   can remove the harmony channel
+//---------------------------------------------------------
+void Part::updateHarmonyChannels(bool checkRemoval)
```

Comment:
The channel management code is something I am not so familiar with, @jthistle and/or @dmitrio95 may wish to look at this.

Reply:
I looked into instrument changes and realized that it doesn't affect the mixer so for the first concern I'm wondering if it would be possible to include these changes in the instrument change dialog. It also might be possible to look into handling these channels in a finer grain setting with your suggestion to include an originating harmony element in the events. 

I would like to clean up how the harmony channel thing works, but for the second point, wouldn't we be able to reuse the same channel for fretboard diagrams/figured bass since they should all be a notation of the same harmonic ideas?